raster/src/test/scala/geotrellis/raster/ArrayTileSpec.scala

```diff
@@ -231,3 +257,42 @@ class ArrayTileSpec extends FunSpec
     }
   }
 }
+
+object ArrayTileSpec {
+  /** BiasedAdd is evaluates `x + NoData` as `x` */
+  case object BiasedAdd extends LocalTileBinaryOp {
+    def op(z1: Int, z2: Int): Int = z1 + z2
+    def op(z1: Double, z2: Double): Double = z1+z2
+
+    def combine(z1: Int, z2: Int): Int =
+      if (isNoData(z1) && isNoData(z2)) NODATA
+      else if (isNoData(z1))
+        z2
+      else if (isNoData(z2))
+        z1
+      else
+        op(z1, z2)
+
+    def combine(z1: Double, z2: Double): Double =
+      if (isNoData(z1) && isNoData(z2)) doubleNODATA
+      else if (isNoData(z1))
+        z2
+      else if (isNoData(z2))
+        z1
+      else
+        op(z1, z2)
+  }
+
+  /** Counts the number of non-NoData cells in a tile */
+  case object CountData {
```

Comment:
Does something like this already exist?

Reply:
No, in general this is a case of https://github.com/locationtech/geotrellis/issues/779 though